### DISCUSSION TITLE
[QUESTION] How to get specific column from multidimensional py::array

### DISCUSSION THREAD

**Discussion by scotie** [pybind/pybind11#3469] [Q&A]
Hi, 

I have a python script with a function that returns a 2-dimentional array and looks something like this:
`[[ 269.83       1391.        ]
 [ 270.17347367 1483.        ]
 [ 270.51702984 1492.        ]
 ...
 [1118.45796377 1537.        ]
 [1118.93168884 1541.        ]
 [1119.40545879 1525.        ]]`

I want to be able to get the whole column 1 ( data[:,1] ), preferably without iterating over the array and convert it to std::vector<double>.

Thank you

**Comment by EricCousineau-TRI** [comment]
Have you tried it out via the Python API via C++?
https://pybind11.readthedocs.io/en/stable/advanced/pycpp/object.html
Somewhat related: https://pybind11.readthedocs.io/en/stable/advanced/pycpp/numpy.html#ellipsis

You should be able to use `slice(None)` as a stand-in for `:`. Then leveraging example from above docs for NumPy:
```
py::array a = ...;
py::array b = a[py::make_tuple(py::slice(nullopt, nullopt, nullopt), 1)]; 
```

**Reply by scotie** [reply]
Thanks, I'll give it a go.

For now, I did something like
`auto r = x.unchecked<2>();

for (py::ssize_t i = 0; i < r.shape(0); i++)
{
       vec1.push_back(r(i, 0));
       vec2.push_back(r(i, 1));
}`

But it seems hacky to me.
